core/src/test/scala/unit/kafka/zk/ZooKeeperTestHarness.scala

```diff
@@ -54,3 +58,40 @@ abstract class ZooKeeperTestHarness extends JUnitSuite with Logging {
     Configuration.setConfiguration(null)
   }
 }
+
+object ZooKeeperTestHarness {
+
+  /**
+   * Verify that a previous test that doesn't use ZooKeeperTestHarness hasn't left behind an unexpected thread
+   */
+  @BeforeClass
+  def setUpClass() {
+    verifyNoUnexpectedThreads()
+  }
+
+  /**
+   * Verify that tests from the current test class using ZooKeeperTestHarness hasn't left behind an unexpected thread
+   */
+  @AfterClass
+  def tearDownClass() {
+    verifyNoUnexpectedThreads()
+  }
+
+  /**
+   * Verifies that threads which are known to cause transient failures in subsequent tests
+   * have been shutdown. These include threads which make connections to brokers and may
+   * cause issues when broker ports are reused (e.g. auto-create topics) as well as threads
+   * which reset static JAAS configuration.
+   */
+  def verifyNoUnexpectedThreads() {
+    val unexpectedThreadNames = Set("controller-event-thread",
+                                    "kafka-producer-network-thread",
+                                    "kafka-coordinator-heartbeat-thread",
+                                    "ZkClient-EventThread")
```

Comment:
Is there something we can do to validate that these thread names haven't changed? We could perhaps introduce constants/methods for our code, at least.